### PR TITLE
Merge main to main-vs-deps

### PR DESCRIPTION
**Pull request by Cosifne**
git fetch --all
git checkout dev/shech/merge-main-to-main-vs-deps
git reset --hard upstream/main-vs-deps
git merge upstream/main

Fix merge conflicts
git commit
git push

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSourceProvider.cs [hunk 1]
```diff
@@ -90,7 +90,7 @@ internal partial class SuggestedActionsSourceProvider : ISuggestedActionsSourceP
             if (textBuffer.IsInLspEditorContext())
                 return null;
 
-            var asyncEnabled = _globalOptions.GetOption(SuggestionsOptions.Asynchronous) ?? _globalOptions.GetOption(SuggestionsOptions.AsynchronousFeatureFlag);
+            var asyncEnabled = _globalOptions.GetOption(SuggestionsOptions.Asynchronous) ?? !_globalOptions.GetOption(SuggestionsOptions.AsynchronousQuickActionsDisableFeatureFlag);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSourceProvider.cs:93]: no.  this isn't right.  it should be an &&, not an ?? in main and main-vs-deps.

File: src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml.cs [hunk 2]
```diff
@@ -96,9 +96,8 @@ public AdvancedOptionPageControl(OptionStore optionStore, IComponentModel compon
             // Quick Actions
             BindToOption(ComputeQuickActionsAsynchronouslyExperimental, SuggestionsOptions.Asynchronous, () =>
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml.cs:97]: in main and main-vs-deps we want to remove the entire BindToOption, and these shouldn't be in the xaml anymore either.

File: src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSourceProvider.cs (hunk 1 above)
Comment (Cosifne) [reply src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSourceProvider.cs:93]: Wired.. It sounds like we are missing some commits in main?

**Comment by Cosifne** [comment]
OK Close this because of this https://github.com/dotnet/roslyn/pull/56858

**Pull request closed without merging**